### ISSUE TITLE
ES6 class appears as unused despite a code below creating an instance of it.

### ISSUE THREAD

**Issue by jpedrosa** [babel/babel-eslint#39]
I was testing ES6 with Babel when I ran into the following problem:

```javascript
function jello(fn) {
  fn();
}
class Hey {
  constructor() {
    this.name = "John";
  }
  sayHello(partingMessage) {
    jello(() => console.log(`Hello, ${this.name}!`));
    console.log(partingMessage);
  }
}
var h = new Hey();
h.sayHello();
```

I get this warning:

  4:6  error  Hey is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

Also, should h.sayHello produce a warning since it's calling it without an argument? Is there an option for that one? :-)

Apparently Espree got a related fix and I also checked out a couple of related bugs. But I couldn't see a code like mine in the samples for the error. Most of them were worried about module exporting instead.

Thanks.

**Comment by sebmck** [comment]
Can you please try `babel-eslint` 1.0.14

**Comment by jpedrosa** [comment]
Sure. I just did. The warning moved to the instance creation line:

  13:12  error  'Hey' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

**Comment by sebmck** [comment]
What version of ESLint?

**Comment by jpedrosa** [comment]
eslint@0.16.0

**Comment by sebmck** [comment]
All the hacks in `babel-eslint` to get linting to work are ~0.15.0 specific.

**Comment by jpedrosa** [comment]
Ouch. :-)

**Comment by sebmck** [comment]
Actually no, ESLint issue. eslint/eslint#1967
